api/controllers/tag.go

```diff
@@ -0,0 +1,201 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/url"
+	"strconv"
+
+	"github.com/quintilesims/layer0/api/tag"
+	"github.com/quintilesims/layer0/client"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+	glob "github.com/ryanuber/go-glob"
+	"github.com/zpatrick/fireball"
+)
+
+type TagController struct {
+	TagStore tag.Store
+}
+
+func NewTagController(t tag.Store) *TagController {
+	return &TagController{
+		TagStore: t,
+	}
+}
+
+func (t *TagController) Routes() []*fireball.Route {
+	return []*fireball.Route{
+		{
+			Path: "/tag",
+			Handlers: fireball.Handlers{
+				"GET":    t.ListTags,
+				"POST":   t.CreateTag,
+				"DELETE": t.DeleteTag,
+			},
+		},
+	}
+}
+
+func (t *TagController) DeleteTag(c *fireball.Context) (fireball.Response, error) {
+	var tag models.Tag
+	if err := json.NewDecoder(c.Request.Body).Decode(&tag); err != nil {
+		return nil, errors.New(errors.InvalidRequest, err)
+	}
+
+	if err := t.TagStore.Delete(tag.EntityType, tag.EntityID, tag.Key); err != nil {
+		return nil, err
+	}
+
+	return fireball.NewJSONResponse(200, nil)
+}
+
+func (t *TagController) CreateTag(c *fireball.Context) (fireball.Response, error) {
+	var tag models.Tag
+	if err := json.NewDecoder(c.Request.Body).Decode(&tag); err != nil {
+		return nil, errors.New(errors.InvalidRequest, err)
+	}
+
+	if err := t.TagStore.Insert(tag); err != nil {
+		return nil, err
+	}
+
+	return fireball.NewJSONResponse(200, tag)
+}
+
+func (t *TagController) ListTags(c *fireball.Context) (fireball.Response, error) {
```

Comment:
Here's a (perhaps) more constructive idea -- let's add some docs to these methods that describe the flow of what's happening similar to what @jparsons04 did in the `provider` layer. I realize this wouldn't be consistent with the lack of docs elsewhere in `controller`, but as this is an outlier, it seems appropriate to call it out as such.

Reply:
added

Reply:
cool, that does help